tools/c7n_azure/tests/test_storage.py

```diff
@@ -42,3 +58,121 @@ def test_value_filter(self):
         })
         resources = p.run()
         self.assertEqual(len(resources), 1)
+
+    @arm_template('storage.json')
+    def test_network_ip_rules_action(self):
+        self.cleanup_policy.run()
+
+        p_add = self.load_policy({
+            'name': 'test-azure-storage-add-ips',
+            'resource': 'azure.storage',
+            'filters': [
+                {'type': 'value',
+                 'key': 'name',
+                 'op': 'glob',
+                 'value_type': 'normalize',
+                 'value': 'cctstorage*'}],
+            'actions': [
+                {'type': 'set-network-rules',
+                 'default-action': 'Deny',
+                 'bypass': 'Logging, Metrics',
+                 'ip-rules': [
+                     {'ip-address-or-range': '11.12.13.14'},
+                     {'ip-address-or-range': '21.22.23.24'}
+                 ]}
+            ]
+        })
+
+        p_add.run()
+
+        p_get = self.load_policy({
+            'name': 'test-azure-storage-enum',
+            'resource': 'azure.storage',
+            'filters': [
+                {'type': 'value',
+                 'key': 'name',
+                 'op': 'glob',
+                 'value_type': 'normalize',
+                 'value': 'cctstorage*'}],
+        })
+
+        resources = p_get.run()
+        self.assertEqual(len(resources), 1)
+        ip_rules = resources[0]['properties']['networkAcls']['ipRules']
+        self.assertEqual(len(ip_rules), 2)
+        self.assertEqual(ip_rules[0]['value'], '11.12.13.14')
+        self.assertEqual(ip_rules[1]['value'], '21.22.23.24')
+        self.assertEqual(ip_rules[0]['action'], 'Allow')
+        self.assertEqual(ip_rules[1]['action'], 'Allow')
+
+        self.cleanup_policy.run()
+        resources = p_get.run()
+        ip_rules = resources[0]['properties']['networkAcls']['ipRules']
+        self.assertEqual(len(ip_rules), 0)
+
+    @arm_template('storage.json')
+    def test_virtual_network_rules_action(self):
+        resources = self.cleanup_policy.run()
```

Comment:
Do you still need this?

Reply:
Yes, I would like to start from a deterministically clean state.

Reply:
I agree with Andy's comment about tearDown(). We can move these tests to a new file test_storage_firewall() if you don't want to apply cleanup to all tests.

Reply:
teardown() runs after a test is run it doesn't guarantee that any test will be run from the required state. MOved the code to the setUp()